### PR TITLE
Support Miniscript as a separate dll.

### PR DESCRIPTION
**Pull request by joemphilips**
* [x] Complete implementation
* [x] Create C# Facade as an extension methods in `NBitcoin` namespace 
* [x] Create `NBitcoin.nuspec` file so that I can include `Miniscript.dll` to NBitcoin's nupkg.
* [ ] Tweak AssemblyInfo and `.fsproj`

### CODE REVIEW

**Comment by NicolasDorier** [comment]
I think you should not manually touch nuspec. The nuspec is generated by MSBuild. There is MSBuild properties to influence the nuget package. I think you should use this.

**Comment by joemphilips** [comment]
I've been trying [a solution 2 in this answer](https://stackoverflow.com/a/45004898/6401031) (and [solutions in the issue linked from there](https://github.com/NuGet/Home/issues/3891).)But this does not work in case we have circular dependency.

I will keep looking. But I'm not sure there is a way to workaround without `.nuspec`

File: NBitcoin/NBitcoin.csproj [hunk 1]
```diff
@@ -20,7 +20,7 @@
 		<EmbedUntrackedSources>true</EmbedUntrackedSources>
 	</PropertyGroup>
 	<PropertyGroup>
-		<TargetFrameworks>net461;net452;netstandard1.3;netstandard1.1;netcoreapp2.1;netstandard2.0</TargetFrameworks>
+		<TargetFrameworks>netstandard1.3;netstandard1.1;netcoreapp2.1;netstandard2.0</TargetFrameworks>
```
Comment (knocte) [inline NBitcoin/NBitcoin.csproj]: why removing net4xx profiles?

Comment (joemphilips) [reply NBitcoin/NBitcoin.csproj]: Uups. By mistake. I will fix

File: NBitcoin/NBitcoin.csproj [hunk 2]
```diff
@@ -93,4 +93,28 @@
 	<PropertyGroup>
 		<DocumentationFile>bin\$(Configuration)\$(TargetFramework)\NBitcoin.xml</DocumentationFile>
 	</PropertyGroup>
+
+	<!-- Include Miniscript dll only if it is ready. -->
```
Comment (joemphilips) [inline NBitcoin/NBitcoin.csproj]: hacky workaround to include Miniscript.dll to `.nupkg` as a separate dll

File: NBitcoin.Miniscript/Program.fs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿// Learn more about F# at http://fsharp.org
+
+open System
+
+[<EntryPoint>]
+let main argv =
+    printfn "Hello World from F#!"
+    0 // return an integer exit code
```
Comment (knocte) [inline NBitcoin.Miniscript/Program.fs]: I'm guessing you can remove this file now

File: NBitcoin.Miniscript.Tests/AssemblyInfo.fs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+// Auto-Generated by FAKE; do not edit
+namespace System
+
+open System.Reflection
+
+[<assembly:AssemblyTitleAttribute("FNBitcoin.Tests")>]
+[<assembly:AssemblyProductAttribute("FNBitcoin")>]
```
Comment (knocte) [inline NBitcoin.Miniscript.Tests/AssemblyInfo.fs]: old names

File: NBitcoin.Miniscript.Tests/AssemblyInfo.fs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+[<assembly:AssemblyMetadataAttribute("GitHash", 
+                                     "bb8964b54bee133e9af64d316dc2cfee16df7f72")>]
+do ()
+
+module internal AssemblyVersionInformation =
+    [<Literal>]
+    let AssemblyTitle = "FNBitcoin.Tests"
+    
+    [<Literal>]
+    let AssemblyProduct = "FNBitcoin"
```
Comment (knocte) [inline NBitcoin.Miniscript.Tests/AssemblyInfo.fs]: old names

File: NBitcoin.Miniscript.Tests/Contract.fs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+                    return v
+                }
+                match res with
+                | Verified s -> Expect.isTrue true "ok"
+                | Failed s -> Expect.isTrue false "failed"
+            else
+                Expect.isTrue true "skip"
+    ]
+
+*)
```
Comment (knocte) [inline NBitcoin.Miniscript.Tests/Contract.fs]: remove file?

File: NBitcoin.Miniscript.Tests/CSharp/MiniscriptPSBTTests.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+{
+	public class MiniscriptPSBTTests
+	{
+		public MiniscriptPSBTTests()
+		{
+		}
+
+        private TransactionSignature DummyKeyFn(PubKey key) => null;
+        public void ShouldSatisfyMiniscript()
+		{
```
Comment (knocte) [inline NBitcoin.Miniscript.Tests/CSharp/MiniscriptPSBTTests.cs]: FYI mixed tabs and spaces, misconfigured editor? ;)

Comment (joemphilips) [reply NBitcoin.Miniscript.Tests/CSharp/MiniscriptPSBTTests.cs]: 😭
I will fix later.

**Comment by nopara73** [comment]
Can you tell me what miniscript is?

**Comment by joemphilips** [comment]
@nopara73 
In short, it is a high-level DSL which compiles to actual bitcoin script, like Ivy.

You can see general overview here. The page also has links to whole other resources.
https://www.youtube.com/watch?v=xOQulGTknQo

I couldn't found reference implementation by Sipa so I'm treating [rust-miniscript](https://github.com/apoelstra/rust-miniscript) as a reference.

**Comment by joemphilips** [comment]
Closing in favor of #684

**Pull request closed without merging**